doc/read-the-docs-site/adr/0002-indexer-resuming-strategy.rst

```diff
@@ -38,187 +36,51 @@ Given this performance issue, we want to define an efficient resuming strategy w
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* It's the indexer responsability to ignore new stable points that are past its
+  current indexing point, if any.
 
-::
-
-             Rollbackable
-              |------|
-  1 2 3 4 5 6 7 8 9 10
-            |--------|
-                A   Tip
-
-Supposing we have a ``securityParam`` of ``4``.
-``A``'s resumable interval provided for the chain-sync protocol is ``[10, 9, 8, 7, 6]``.
-Reminder that we return a maximum of ``securityParam + 1`` number of resumable points for any indexer.
-The chain-sync protocol will try each of these points and identify the first one which is known by the local node.
-As rollbacks can occur between points ``[7..10]`` after the indexer was stopped, the points ``[7..10]`` provided by the indexer *could* be invalid.
-Thus, if any of those rollbackable points fail, we can at least guaranty that the chain-sync protocol will resume at point ``6``.
-That is unless the node database was deleted and the node re-sync did not get past point ``6`` in a scenario such as:
-
-::
-
-  1 2 3 4 5 6 7 8 9 10 11 12 13
-      |     |--------|
-     Tip        A
-
-In that case, we will just log a runtime error saying that the provided resuming points don't appear in the chain.
-Then, the user will have to wait for the node to sync up.
-
-``A`` and ``B`` are resuming at different points
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-::
-
-                      Rollbackable
-                        |-----|
-  1 2 3 4 5 6 7 8 9 10 11 12 13
-    |---|                     |
-      B                      Tip
-  |---|
-    A
-
-Given our resuming point selection, the resumable interval provided for the chain-sync protocol is ``[3, 2, 1]``.
-``A`` will start syncing from point ``4``, while ``B`` will drain/ignore points ``3`` and ``4`` and start syncing from point ``5``.
-
-Let's take a similar scenario:
+* We will start a single node chain-sync client for all indexers instead of one node chain sync client per indexer.
 
-::
+Argument
+--------
 
-                      Rollbackable
-                        |-----|
-  1 2 3 4 5 6 7 8 9 10 11 12 13
-    |---|                     |
-      B                      Tip
-  |---------|
-       A
+Our different implementations of the resuming so far tried to restart at thhe
+latest common synchronisation points of all the indexers, even if this point was
+a volatile one.
+The consequence of it was a complex logic to (1) find a valid sync point, (2)
+deal with the potential rollbacks once the chainpoint was chosen.
 
-Given our resuming point selection, the resumable interval provided for the chain-sync protocol is ``[4, 3, 2]``.
-``B`` will start syncing from point ``5``, while ``A`` will drain/ignore points ``5`` and ``6`` and start syncing from point ``7``.
+Sticking to stable blocks drastically simplified the resuming logic, as we don't
```

Comment:
** suggestion**

Scoping the resume to stable blocks only drastically simplified the resume logic.  This design removed the complex logic of determining whether rollbacks occurred while the indexer(s) were offline. This simplification came at the cost of potentially re-indexing the volatile blocks. We have determined, this is a small cost to pay for the gained simplifications.